bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BigtableAsyncUtilities.java

```diff
@@ -114,10 +141,32 @@
           StatusRuntimeException.class,
           RetryingRpcFunction.create(retryOptions, request, rpc),
           executorService);
+    } else {
+      return listenableFuture;
     }
-    return listenableFuture;
   }
 
+  /**
+   * Returns the result of calling {@link Future#get()} interruptably on a task known not to throw a
+   * checked exception.
+   *
+   * <p>If interrupted, the interrupt is restored before throwing an exception..
+   *
+   * @throws java.util.concurrent.CancellationException
+   *     if {@code get} throws a {@code CancellationException}.
+   * @throws io.grpc.StatusRuntimeException if {@code get} throws an {@link ExecutionException}
+   *     or an {@link InterruptedException}.
+   */
+  public static <V> V getUnchecked(Future<V> future) {
+    try {
+      return future.get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw Status.CANCELLED.withCause(e).asRuntimeException();
+    } catch (ExecutionException e) {
+      throw Status.fromThrowable(e).asRuntimeException();
```

Comment:
This code was copied from gRPC.  Status.fromThrowable(e) looks at the exception to figure out what the status should be.  It looks to see if it's a gRPC exception that contains a status, and if so, use that status.